features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/WireframeMapper.kt

```diff
@@ -22,9 +24,18 @@ interface WireframeMapper<in T : View, out S : MobileSegment.Wireframe> {
      * Maps a [View] to a [List<Wireframe>] in order to be rendered in the Session Replay player.
      * @param view as the [View] instance that will be mapped
      * @param mappingContext in which we provide useful information regarding the current
-     * system state.
+     * @param asyncJobStatusCallback a callback that can be called when the mapper starts or
+     * finishes processing an async job. By calling this whenever a job started (in the main thread)
+     * and finished (in the background thread) you will make sure that the `List<Wireframe>` will
+     * not be consumed until all the wireframes are updated by the async jobs. The main reason we
+     * needed this functionality was to be able to unblock the main thread whenever we needed
+     * heavy operations like image capture or serialisation.
```

Comment:
I would re-word this sentence about why it is needed. Not to say "we needed this functionality", but something like "it can be used to offload heavy work from the calling thread (like main) ... ", in such fashion.